extensions/blocks/amazon/edit.js

```diff
@@ -1,23 +1,261 @@
+					</Notice>
+				)
+			}
+		>
+			<form>
+				<FormTokenField
+					value={ asin }
+					suggestions={ suggestions }
+					onInputChange={ onInputChange }
+					maxSuggestions={ 10 }
+					label={ __( 'Products', 'jetpack' ) }
+					onChange={ onChange }
+				/>
+				<Button isSecondary isLarge type="submit">
+					{ __( 'Preview', 'jetpack' ) }
+				</Button>
+			</form>
+		</Placeholder>
+	);
 
-export default AmazonEdit;
+	const inspectorControls = (
+		<InspectorControls>
+			{ asin && (
+				<>
+					<PanelBody title={ __( 'Promotion Settings', 'jetpack' ) }>
+						<ToggleControl
+							label={ __( 'Show Image', 'jetpack' ) }
+							checked={ showImage }
+							onChange={ () => setAttributes( { showImage: ! showImage } ) }
+						/>
+						<ToggleControl
+							label={ __( 'Show Title', 'jetpack' ) }
+							checked={ showTitle }
+							onChange={ () => setAttributes( { showTitle: ! showTitle } ) }
+						/>
+						<ToggleControl
+							label={ __( 'Show Author/Seller', 'jetpack' ) }
+							checked={ showSeller }
+							onChange={ () => setAttributes( { showSeller: ! showSeller } ) }
+						/>
+						<ToggleControl
+							label={ __( 'Show Price', 'jetpack' ) }
+							checked={ showPrice }
+							onChange={ () => setAttributes( { showPrice: ! showPrice } ) }
+						/>
+						<ToggleControl
+							label={ __( 'Show Purchase Button', 'jetpack' ) }
+							checked={ showPurchaseButton }
+							onChange={ () => setAttributes( { showPurchaseButton: ! showPurchaseButton } ) }
+						/>
+					</PanelBody>
+					<PanelColorSettings
+						title={ __( 'Color Settings', 'jetpack' ) }
+						colorSettings={ [
+							{
+								value: backgroundColor,
+								onChange: newBackgroundColor =>
+									setAttributes( { backgroundColor: newBackgroundColor } ),
+								label: __( 'Background Color', 'jetpack' ),
+							},
+							{
+								value: textColor,
+								onChange: newTextColor => setAttributes( { textColor: newTextColor } ),
+								label: __( 'Text Color', 'jetpack' ),
+							},
+							{
+								value: buttonAndLinkColor,
+								onChange: newButtonAndLinkColor =>
+									setAttributes( { buttonAndLinkColor: newButtonAndLinkColor } ),
+								label: __( 'Button & Link Color', 'jetpack' ),
+							},
+						] }
+					>
+						{
+							<ContrastChecker
+								{ ...{
+									isLargeText: false,
+									textColor: textColor,
+									backgroundColor: backgroundColor,
+								} }
+							/>
+						}
+					</PanelColorSettings>
+				</>
+			) }
+		</InspectorControls>
+	);
+
+	const blockPreview = () => {
+		const {
+			title,
+			authors,
+			artists,
+			actors,
+			manufacturer,
+			detailPageUrl,
+			listPrice,
+			imageUrlMedium,
+			imageWidthMedium,
+			imageHeightMedium,
+			authorshipInfo,
+		} = data.products.filter( productDataItem => productDataItem.asin === asin )[ 0 ];
+
+		// TODO - we should be able to get this from API in a neater way once we have access
+		const seller = authors.length
```

Comment:
We can look at it some more when we get access to the API, but we're making assumptions about the author or artists superseding the manufacturer for display, and that might not be the case. Hopefully, we can set some sensible defaults like that, otherwise, we might have to get them to pick a field, and that feels messy.

Reply:
Yeah, this isn't the right way to do it, definitely, I'm just waiting to see what the API actually returns...